modules/FiniteSetsExt.tla

```diff
@@ -2,60 +2,59 @@
 LOCAL INSTANCE Naturals
 LOCAL INSTANCE FiniteSets
 LOCAL INSTANCE Functions
+LOCAL INSTANCE Folds
 
-
-
-(***************************************************************************)
-(* Starting from base, apply op to f(x), for all x \in S, in an arbitrary  *)
-(* order. It is assumed that op is associative and commutative.            *)
-(***************************************************************************)
-FoldMap(op(_,_), base, f(_), S) ==
-  LET iter[s \in SUBSET S] ==
-        IF s = {} THEN base
-        ELSE LET x == CHOOSE x \in s : TRUE
-             IN  op(f(x), iter[s \ {x}])
-  IN  iter[S]
-
-(*
-   Intuitive semantics:
-
-   Initialize a temporary variable `accum` with the value of `base`.  Iterate
-   over the elements in set in a deterministic but unknown order, update the
-   value of `accum` to the value of `op(accum, x`), where `x` is the current
-   value of the iterator. The result of `FoldSet` is the computed value of
-   `accum`.
- *)  
 FoldSet(op(_,_), base, set) ==
-  (*
-    This is an implementation in terms of previously defined operators.
-    A tool is free to define its own, more efficient, implementation.
-   *) 
-(*  ReduceSet(op, set, base) *)
-  (* sm: alternative, equivalent definition *)
-  FoldMap(op, base, LAMBDA x : x, set)
-
-(***************************************************************************)
-(* Fold over a function (or sequence).                                     *)
-(***************************************************************************)
-FoldFunction(op(_,_), base, fun) == 
-  FoldMap(op, base, LAMBDA i : fun[i], DOMAIN fun)
-
-
-
-(***************************************************************************)
-(* Fold over a subset of the domain of a function -- corresponds to Reduce *)
-(***************************************************************************)
-FoldFunctionOnSet(op(_,_), base, fun, set) ==
-  LET Squeeze(f, S) == [ s \in S |-> f[s]  ]
-  IN
-  FoldFunction(op, base, Squeeze(fun, set))
-
+   (*************************************************************************)
+   (* Fold op over the elements of set using base as starting value.        *)
+   (*                                                                       *)
+   (* Intuitive semantics:                                                  *)
+   (* Initialize a temporary variable `accum` with the value of `base`.     *)
+   (* Iterate over the elements in set in a deterministic but unknown       *)
+   (* order pdate the value of `accum` to the value of `op(accum, x`),      *)
+   (* where `x` s the current value of the iterator.                        *)
+   (*                                                                       *)
+   (* Example:                                                              *)
+   (*         FoldSet(LAMBA x,y : x + y, 0 .. 10, 0) = 55                   *)
```

Comment:
Mismatch: Example mentions `FoldSet` but operator name is `MapThenFoldSet`.

Reply:
Isn't this in the definition of FoldSet by MapThenFoldSet ?